### PR TITLE
fix(nuxt): preserve percent-encoding in navigateTo

### LINKED ISSUE #33273: `navigateTo` does not preserve percent-encoding in its query param
### Environment

- Operating System: `Linux`
- Node Version:     `v24.7.0`
- Nuxt Version:     `4.1.2`
- CLI Version:      `3.28.0`
- Nitro Version:    `2.12.6`
- Package Manager:  `pnpm@10.15.1`
- Builder:          `-`
- User Config:      `compatibilityDate`, `devtools`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

https://github.com/phoenix-ru/nuxt-navigate-to-fullpath-reproduction

Visit the `/` page and observe server logs.

### Describe the bug

The bug was initially found in https://github.com/sidebase/nuxt-auth/pull/1057

We use a custom implementation of `navigateToAuthPage` which is mirrored from the official `navigateTo` helper.

When on the server, `navigateTo` waits for all the middleware to be done and only then calls its `redirect` function:
https://github.com/nuxt/nuxt/blob/dc69e26c5b9adebab3bf4e39417288718b8ddf07/packages/nuxt/src/app/composables/router.ts#L214-L219

When debugging the redirects, I noticed that `redirect` was never actually called from inside the middleware because `final.fullPath === fullPath` check is never true (also in [our implementation](https://github.com/sidebase/nuxt-auth/blob/c806709d561939951edc80dd536ec0536f3340ed/src/runtime/composables/authjs/utils/navigateToAuthPage.ts#L60-L67)).

[Adding verbose logging](https://github.com/phoenix-ru/nuxt-navigate-to-fullpath-reproduction/blob/master/patches/nuxt.patch) to `navigateTo`, I discovered the following:
```js
Scheduling router.afterEach
Redirect was called { fullPath: '/redirect?callback=/other',
  response: undefined,
  isExternal: false,
  inMiddleware: false }
router.afterEach was called { fullPath: '/redirect?callback=%2Fother',
  'final.fullPath': '/redirect?callback=/other',
  'final.fullPath === fullPath': false,
  isExternal: false,
  inMiddleware: true }
```

To be exact:
- `fullPath` (the parameter provided to `navigateTo`) was being transformed to percent-decoded string inside Vue Router;
- `navigateTo` was for some reason being called twice: once in middleware which scheduled `router.afterEach` and once outside middleware which seems to have rewritten the `fullPath`.

### Additional context

I have checked the resolution in `vue-router` and it does not seem to be a problem with `router.resolve` (at least based on the experimental (?) [route resolver spec](https://github.com/vuejs/router/blob/1885046dd5081a80c26a58ef94381c1b0e9ace2c/packages/router/src/experimental/route-resolver/resolver-fixed.spec.ts#L897-L913)).

I think the double-calling of `navigateTo` might be the culprit

### Logs

```shell-script

```

### PR DESCRIPTION
**Pull request by onmax**
Closes #33273

## Summary
`navigateTo` compares raw `fullPath` with vue-router's percent-encoded `final.fullPath`, causing redirect to fail. Fix: normalize both via `encodeURL()` before comparing.

## Reproduction (nuxt@4.3.0)

| | Link | Expected |
|---|---|---|
| Bug | [nuxt-33273](https://stackblitz.com/github/onmax/repros/tree/repro/nuxt-33273/nuxt-33273?startScript=dev) | ❌ Redirect fails |
| Fix | [nuxt-33273-fix](https://stackblitz.com/github/onmax/repros/tree/repro/nuxt-33273/nuxt-33273-fix?startScript=dev) | ✅ Redirect works |

Visit `/trigger` → redirects to `/target?callback=/other`

### CODE REVIEW

**Comment by onmax** [comment]
@danielroe reworked PR based on review feedback:

- Test now uses `https://example.com` to actually test percent-encoding (contains `:` and `//`)
- Added `encodeURL()` normalization in router.ts lines 155 & 171 for consistent comparison

Ready for review

File: packages/nuxt/src/pages/runtime/plugins/router.ts [hunk 1]
```diff
@@ -165,7 +165,7 @@ const plugin: Plugin<{ router: Router }> = defineNuxtPlugin({
       await nuxtApp.runWithContext(() => showError(error))
     }
 
-    const resolvedInitialRoute = import.meta.client && initialURL !== router.currentRoute.value.fullPath
+    const resolvedInitialRoute = import.meta.client && encodeURL(initialURL) !== encodeURL(router.currentRoute.value.fullPath)
```
Comment (posva) [inline packages/nuxt/src/pages/runtime/plugins/router.ts:168]: FYI running encodeURL is probably not the correct approach as it will create double encoding for Vue router. I haven’t checked the original issue but I think this might hide a different issue somewhere else. Worth trying URLs with special characters (anything not ascii like ñ or 嗨) in the path and other sections of the URL

Comment (onmax) [reply packages/nuxt/src/pages/runtime/plugins/router.ts:168]: @posva Thanks for the heads up. Did some digging into Vue Router's encoding:

 **Vue Router's `encodeQueryValue()` keeps these unencoded in query values:**
 - `:`, `/`, `[`, `]`, `{`, `}`, `^`, backtick
 - Space -> `+`, literal `+` -> `%2B`

The problem is that `encodeURL()` uses `new URL()` which preserves input encoding without normalizing. So these comparisons fail:

 ```js
 encodeURL('?time=10%3A30') !== encodeURL('?time=10:30')     // %3A vs :
 encodeURL('?url=http%3A%2F%2F...') !== encodeURL('?url=http://...')
 encodeURL('?q=hello+world') !== encodeURL('?q=hello%20world')  // + vs %20
```

When `initialURL` (from HTTP request) has encoded chars and Vue Router's `fullPath` has literals, the comparison breaks.

I think we have two options:

 1. Decode before comparing - simple `decodeURIComponent()` on both sides
 2. Match Vue Router's encoding - decode then re-encode using Vue Router's scheme

 ption 1 is simpler but loses encoding info. Option 2 is more correct but requires replicating Vue Router's encoding logic.

What's your recommendation? Also happy to add failing tests to document these edge cases.

Comment (posva) [reply packages/nuxt/src/pages/runtime/plugins/router.ts:168]: I actually thought encodeURL was encodeURI 😆 
I haven't dived into the issue to see where it comes from exactly, that would take more time but I still thought it would be helpful to drop a message
